.github/workflows/tests.yml

```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go: ['1.19', '1.18', '1.17', '1.16', '1.15', '1.14']
+        go: ['1.20', '1.19', '1.18', '1.17']
```

Comment:
I could not from the current discussions captured here confirm what the decision was here? I.e. Why do we start from Go 1.17 for tests specifically (and not Go 1.14, or Go 1.18)?

I did a quick test, and the following is possible, if we want to do something more magical (and complex :p ). Right now it looks a bit pointless as all 3 cases pick Go 1.20.

```
jobs:
  run-checks:

    runs-on: ubuntu-latest
    strategy:
      matrix:
        include:
          - comment: 'go-latest'
            go_ver: '>=1.20'
            go_mod: ''
            latest: true
          - comment: '1.20'
            go_ver: '1.20'
            go_mod: ''
            latest: false
          - comment: 'go_mod'
            go_ver: ''
            go_mod: 'go.mod'
            latest: false

    name: run-checks-with-${{ matrix.comment }}

    steps:
      - name: Checkout Code
        uses: actions/checkout@v2
        with:
          fetch-depth: 1

      - name: Setup Go Version
        uses: actions/setup-go@v4
        with:
          go-version: ${{ matrix.go_ver }}
          go-version-file: ${{ matrix.go_mod }}
          check-latest: ${{ matrix.latest }}

      - name: Checks
        run: |
          VERBOSE=1 ./run-checks
```

Reply:
(On the more complex selection - I'm not sure it's gaining us much right now, other than some head scratching for people maintaining this repo! ;-))